### ISSUE TITLE
Don't report ILOffset as Line number

### ISSUE THREAD

**Issue by bruno-garcia** [getsentry/raven-csharp#258]
In case the Line number is not available, currently we fallback to IL Offset.
The problem with this approach is that it becomes misleading and we've had support issues opened because of this.

- [ ] Do not implement a fallback in case Line Number is not available
- [ ] IL Offset can be reported via the field: `instruction_addr`. The value is a hex string currently limited to <=53bit

Docs: https://docs.sentry.io/clientdev/interfaces/stacktrace/

**Comment by jviolas** [comment]
Will this have any impact on how Sentry is grouping events when no line number is found ?

**Comment by bruno-garcia** [comment]
Hi @jviolas! It will not affect grouping as we do not use the line number for it.

**Comment by jviolas** [comment]
Ok. Thanks. That's good to know.
I'm currently testing and integrating Sentry in a .Net WPF desktop app and I've had some issues with grouping stacktraces on other error reporting services, that's why I was wondering.

**Comment by bruno-garcia** [comment]
It's a good point. How's sentry working out for you?
We'd really appreciate your feedback!

**Comment by jviolas** [comment]
For now, great. We tested it with some various crashes to see how they're were handled and grouped. Even with some surrounding code changing to see if new crash occurrence was still group with previous ones. So far, it's good and we've decided to go with it. Also for an iOS/Android app with cordova.
The Jira integration doesn't quite fit our needs (we would prefer a two way sync for issue status) but we will manage this with web hooks, REST API and a server between that.

As for the SharpRaven .Net SDK. I've been able to do almost everything I wanted but I had to checkout the repo and see the code myself to better understand some part because documentation was not very detailed (didn't have to search, the code is very well documented !!). And I'll have to find a way to manage offline exception.

**Comment by bruno-garcia** [comment]
@jviolas Thanks for the detailed feedback! We're working on improving the experience for .NET devs further. That includes a new API to support future use cases and also unify the API across SDKs. With a unified API it'll also be easier to document the SDKs, too. We'll be focusing on that soon.

It'll also contain event persistence to allow offline scenarios. That'll be default for Xamarin integrations (iOS/Android).
We're working on the new, unified API on a [different repo](https://github.com/getsentry/sentry-dotnet), in case you'd like to get involved.

**Comment by jviolas** [comment]
@bruno-garcia Yes I've already seen the new repo and watching it. Don't hesitate to ping me if you need some test/feedback on the new SDK, especially in the context of a .NET WPF desktop app (not Xamarin or UWP). I'll be happy to help if Ican (and have time, that's the most difficult part...)
